test/lib/ansible_test/_data/requirements/sanity.import.plugin.in

```diff
@@ -1,2 +1,3 @@
 jinja2  # ansible-core requirement
 pyyaml  # ansible-core requirement
+packaging  # ansible-core requirement
```

Comment:
Instead of adding `packaging` here, guard the import, since `packaging` is not considered a "public" dependency. We're already doing this for other uses of `packaging`, such as:

https://github.com/ansible/ansible/blob/681c32ec65ec5ae104663464ac69f890d799c2e9/lib/ansible/galaxy/collection/__init__.py#L33-L35

See: https://docs.ansible.com/ansible/latest/dev_guide/testing/sanity/import.html

Reply:
https://github.com/ansible/ansible/pull/76308 separated our requirements into two categories:

> Public dependencies of ansible-core are:
> 
>   * Jinja2
>   * PyYAML
>   * MarkupSafe (as a dependency of Jinja2)

Public dependencies are used widely in the ansible-core code and plugins can import them without handling `ImportError`. They're dependencies that are heavily used throughout the ansible-core code.

The remaining dependencies are essentially implementation details behind specific features of ansible-core and plugins should not assume they will be present. They are handled by the `import` sanity test the same as any other import from outside the Python standard library. These are currently:

* pycparser
* cffi
* cryptography
* pyparsing
* packaging
* resolvelib